routes/oauth.js

```diff
@@ -52,13 +52,13 @@ router.get("/redirect", (req, res) => {
         headers: {
           Authorization: `Bearer ${user.accessToken}`,
         },
-      }).then((data) => {
-        const email = JSON.parse(data.body).email;
-        gEmails.add(email);
-        res.send(`Registered ${email} for breach alerts. You may now close this window/tab.`);
       });
-    })
-    .catch((err) => {
+    }).then((data) => {
+      return Subscribers.addUser(JSON.parse(data.body).email);
+    }).then(() => {
+      res.send(`Registered ${email} for breach alerts. You may now close this window/tab.`);
```

Comment:
I'm a bit lost here... I'm not sure what, if anything, the `${email}` refers to here. Previously it was a stored copy of `JSON.parse(data.body).email`, but that doesn't seem to be getting stored or passed in the promise chain, that I can immediately see.

Reply:
And from my limited knowledge of `Subscribers.addUser()`, it should return an empty object on success, so we can't rely on returning the `.addUser()` response. We may need to cache the `JSON.parse(data.body).email` value and return that in the promise chain.